Library/Homebrew/os/linux/elf.rb

```diff
@@ -89,13 +89,16 @@ def with_interpreter?
   def dynamic_elf?
     return @dynamic_elf if defined? @dynamic_elf
 
-    @dynamic_elf = if which "readelf"
-      Utils.popen_read("readelf", "-l", to_path).include?(" DYNAMIC ")
-    elsif which "file"
-      !Utils.popen_read("file", "-L", "-b", to_path)[/dynamic|shared/].nil?
-    else
-      raise "Please install either readelf (from binutils) or file."
-    end
+    @dynamic_elf =
+      if HOMEBREW_READELF_RB
+        patcher.instance_variable_get("@elf").segment_by_type(:DYNAMIC).present?
+      elsif which "readelf"
+        Utils.popen_read("readelf", "-l", to_path).include?(" DYNAMIC ")
+      elsif which "file"
+        !Utils.popen_read("file", "-L", "-b", to_path)[/dynamic|shared/].nil?
+      else
+        raise "Please install either readelf (from binutils) or file."
```

Comment:
Possibly `odie` instead of `raise` here? Not sure.

Reply:
that was existing code. change it ?

Reply:
Nope, let's leave it if it's preexisting.